### ISSUE TITLE
How to define `cube()` in Typescript

### ISSUE THREAD

**Issue by tbadlov** [statsbotco/cube.js#69]
Hello I am trying to define a cube schema in a typescript based project but I cannot just define `cube()` files in and also cannot define a `.js` files.

Any documentation or support for TypeScript?

Thanks!

**Comment by dtslvr** [comment]
Hi @tbadlov, there is no TypeScript support yet, please see #28.

**Comment by paveltiunov** [comment]
@tbadlov Yep. @dtslvr right. There's no TypeScript support yet. But could you please elaborate on your use case? Do you want to define Cube.js schema using TypeScript?

**Comment by tbadlov** [comment]
That’s correct. Just writing ‘cube()’ will not compile with TypeScript. It needs to be defined.

**Comment by paveltiunov** [comment]
Yep. Right. We'd need to add TypeScript support in Cube.js schema transpiler then so it produces correct TypeScript.

**Comment by Grzesie2k** [comment]
@tbadlov You can create `*.d.ts` (eg. `types.d.ts`) file in Your project and add simple definition for `cube` function.
```ts
declare function cube(name: string, definition: any): void;
```
https://www.typescriptlang.org/docs/handbook/declaration-files/by-example.html#global-variables

**Comment by onexdata** [comment]
Hello. I'm very interested in using TypeScript with Cube.js. I converted the dashboard-app but noticed Cube.js keeps looking for app.js instead of app.tsx now.  I also cannot use the explorer to port more components.

Is there a list of remaining items that need to be achieved in order to get official typescript support for Cube.js?

**Comment by paveltiunov** [comment]
@onexdata Hey Nick! Thanks for following up here! This issue specifically refers to TypeScript support in Cube.js Schema files. I think it makes sense to create separate issue for Playground dashboard TypeScript support.
